src/Application/Engine.cpp

```
@@ -242,12 +242,12 @@ void Engine::startThreadMonitor() {
 
             if (now - lastHeartbeat >= std::chrono::milliseconds(AppConst::MAX_MAIN_THREAD_TIMEOUT)) {
                 // If the time elapsed between now and the last heartbeat exceeds the maximum timeout, signal to all worker threads that the main thread has halted
-                if (!g_appCtx.MainThread.isHalted.load()) {
+                if (!g_appCtx.MainThread.isHalted.load(std::memory_order_relaxed)) {
                     m_eventDispatcher->dispatch(UpdateEvent::ApplicationStatus{
```

Cppcheck (2.20.0)

[style] The function 'severity' is never used.

(unusedFunction)